packages/scoutgame-ui/src/providers/DraftProvider.tsx

```diff
@@ -0,0 +1,159 @@
+'use client';
+
+import { saveDraftTransactionAction } from '@packages/scoutgame/drafts/saveDraftTransactionAction';
+import { scoutgameMintsLogger } from '@packages/scoutgame/loggers/mintsLogger';
+import { useAction } from 'next-safe-action/hooks';
+import type { ReactNode } from 'react';
+import { createContext, useCallback, useContext, useMemo, useState } from 'react';
+import { toast } from 'sonner';
+import type { Address } from 'viem';
+import { useSendTransaction } from 'wagmi';
+
+import { useUser } from './UserProvider';
+
+type DraftTransactionInput = {
+  txData: {
+    to: Address;
+    data: `0x${string}`;
+    value: bigint;
+  };
+  txMetadata: {
+    fromAddress: Address;
+    sourceChainId: number;
+    developerId: string;
+    bidAmount: number;
+    season: string;
+  };
+};
+
+type DraftContext = {
+  isExecutingTransaction: boolean;
+  isSavingDraftTransaction: boolean;
+  savedDraftTransaction: boolean;
+  transactionHasSucceeded: boolean;
+  draftError?: string;
+  sendDraftTransaction: (input: DraftTransactionInput) => Promise<unknown>;
+  clearDraftSuccess: () => void;
+  draftSuccess: boolean;
+};
+
+export const DraftContext = createContext<Readonly<DraftContext | null>>(null);
+
+export function DraftProvider({ children }: { children: ReactNode }) {
+  const { refreshUser } = useUser();
+  const { sendTransactionAsync } = useSendTransaction();
+
+  const [draftSuccess, setDraftSuccess] = useState(false);
+
+  const {
+    executeAsync: saveDraftTransaction,
+    isExecuting: isSavingDraftTransaction,
+    hasSucceeded: savedDraftTransaction,
```

Comment:
suggest renaiming this to "draftTransactionSaved" cuz i keep getting it mixed up with saveDraftTransaction